crates/core/tedge_agent/src/agent.rs

```diff
@@ -628,21 +629,27 @@ impl SmAgent {
 }
 
 #[cfg(test)]
-fn get_restart_command() -> Command {
+fn get_restart_operation_command_vec() -> Vec<Command> {
+    let mut vec = vec![];
     // running `echo 6` with no sudo
     let mut command = std::process::Command::new(INIT_COMMAND);
     command.arg("6");
-    command
+    vec.push(command);
+    vec
 }
 
 #[cfg(not(test))]
-fn get_restart_command() -> Command {
+fn get_restart_operation_command_vec() -> Vec<Command> {
+    let mut vec = vec![];
     // sync first
-    let _process_result = std::process::Command::new("sudo").arg("sync").status();
+    let mut sync_command = std::process::Command::new("sudo");
+    sync_command.arg("sync");
+    vec.push(sync_command);
     // running `sudo init 6`
     let mut command = std::process::Command::new("sudo");
     command.arg(INIT_COMMAND).arg("6");
-    command
+    vec.push(command);
+    vec
```

Comment:
Minor - you can avoid a `mut` here.

```suggestion
vec![sync_command, command]